homeassistant/components/xiaomi_miio/alarm_control_panel.py

```diff
@@ -76,6 +77,23 @@ async def _try_command(self, mask_error, func, *args, **kwargs):
         except DeviceException as exc:
             _LOGGER.error(mask_error, exc)
 
+    @callback
+    def alarm_callback(self, action, params):
+        """Push from gateway."""
+        if action == "alarm_triggering":
+            self._attr_state = STATE_ALARM_TRIGGERED
+            self.schedule_update_ha_state()
```

Comment:
Use async api in async context.
```suggestion
            self.async_write_ha_state()
```